Compilation warnings

Hello,

I compiled your code on `Ubuntu 16.04.3 LTS` and I saw many warnings (I am able to fix it, so stay tuned):

```bash
celestian@vega:~/demoinfogo-linux$ make
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofile.cpp -o src/demofile.o
src/demofile.cpp: In member function ‘bool CDemoFile::Open(const char*)’:
src/demofile.cpp:150:56: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
   fread( &m_DemoHeader, 1, sizeof( m_DemoHeader ), fp );
                                                        ^
src/demofile.cpp:168:45: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
   fread( &m_fileBuffer[ 0 ], 1, Length, fp );
                                             ^
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/steammessages.proto
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/cstrike15_gcmessages.proto
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/netmessages_public.proto
mkdir -p src/generated_proto
protobuf-2.5.0/src/protoc --proto_path=./src --proto_path=protobuf-2.5.0/src --cpp_out=./src/generated_proto src/cstrike15_usermessages_public.proto
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofiledump.cpp -o src/demofiledump.o
In file included from src/demofiledump.h:30:0,
                 from src/demofiledump.cpp:27:
src/demofilepropdecode.h: In member function ‘void Prop_t::Print(int)’:
src/demofilepropdecode.h:88:10: warning: enumeration value ‘DPT_NUMSendPropTypes’ not handled in switch [-Wswitch]
   switch ( m_type )
          ^
src/demofiledump.cpp: In function ‘void ParseStringTableUpdate(CBitRead&, int, int, int, int, int, bool)’:
src/demofiledump.cpp:678:17: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if ( nBytes > sizeof( tempbuf ) )
                 ^
src/demofiledump.cpp:728:73: warning: format ‘%s’ expects argument of type ‘char*’, but argument 5 has type ‘const void*’ [-Wformat=]
     printf( " %d, %s, %d, %s \n", entryIndex, pEntry, nBytes, pUserData );
                                                                         ^
src/demofiledump.cpp:595:6: warning: unused variable ‘lastDictionaryIndex’ [-Wunused-variable]
  int lastDictionaryIndex = -1;
      ^
src/demofiledump.cpp: In function ‘CSVCMsg_SendTable* GetTableByClassID(uint32)’:
src/demofiledump.cpp:835:38: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if ( s_ServerClasses[ i ].nClassID == nClassID )
                                      ^
src/demofiledump.cpp: In function ‘void FlattenDataTable(int)’:
src/demofiledump.cpp:1003:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (prop->priority() == priority || (priority == 64 && (SPROP_CHANGES_OFTEN & prop->flags()))) 
                          ^
src/demofiledump.cpp: In function ‘void PrintNetMessage(CDemoFileDump&, const void*, int) [with T = CSVCMsg_PacketEntities; int msgType = 26]’:
src/demofiledump.cpp:1159:7: warning: unused variable ‘nBaseline’ [-Wunused-variable]
   int nBaseline = msg.baseline();
       ^
src/demofiledump.cpp:1160:8: warning: unused variable ‘bUpdateBaselines’ [-Wunused-variable]
   bool bUpdateBaselines = msg.update_baseline();
        ^
src/demofiledump.cpp: In function ‘bool ParseDataTable(CBitRead&)’:
src/demofiledump.cpp:1453:7: warning: unused variable ‘type’ [-Wunused-variable]
   int type = buf.ReadVarInt32();
       ^
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demoinfogo.cpp -o src/demoinfogo.o
In file included from src/demofiledump.h:30:0,
                 from src/demoinfogo.cpp:25:
src/demofilepropdecode.h: In member function ‘void Prop_t::Print(int)’:
src/demofilepropdecode.h:88:10: warning: enumeration value ‘DPT_NUMSendPropTypes’ not handled in switch [-Wswitch]
   switch ( m_type )
          ^
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofilebitbuf.cpp -o src/demofilebitbuf.o
src/demofilebitbuf.cpp: In member function ‘float CBitRead::ReadBitFloat()’:
src/demofilebitbuf.cpp:691:32: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  return *( ( float * ) &nvalue );
                                ^
g++ -m32 -O3 -Wall -fpermissive --std=c++0x -Iprotobuf-2.5.0/src -c src/demofilepropdecode.cpp -o src/demofilepropdecode.o
In file included from src/demofiledump.h:30:0,
                 from src/demofilepropdecode.cpp:25:
src/demofilepropdecode.h: In member function ‘void Prop_t::Print(int)’:
src/demofilepropdecode.h:88:10: warning: enumeration value ‘DPT_NUMSendPropTypes’ not handled in switch [-Wswitch]
   switch ( m_type )
          ^
g++ -Iprotobuf-2.5.0/src src/demofile.o src/demofiledump.o src/demoinfogo.o src/demofilebitbuf.o src/demofilepropdecode.o src/generated_proto/steammessages.pb.cc src/generated_proto/cstrike15_gcmessages.pb.cc src/generated_proto/netmessages_public.pb.cc src/generated_proto/cstrike15_usermessages_public.pb.cc protobuf-2.5.0/src/.libs/libprotobuf.a -m32 -pthread -static -o demoinfogo
```

--- a/src/demofile.cpp
+++ b/src/demofile.cpp
@@ -130,6 +130,9 @@ bool CDemoFile::Open( const char *name )
 {
 	Close();
 
+	size_t read_size;
+	(void) read_size;
+
 	FILE *fp = NULL;
 	fp = fopen( name, "rb" );
 	if( fp )
@@ -147,7 +150,7 @@ bool CDemoFile::Open( const char *name )
 			return false;
 		}
 
-		fread( &m_DemoHeader, 1, sizeof( m_DemoHeader ), fp );
+		read_size = fread( &m_DemoHeader, 1, sizeof( m_DemoHeader ), fp );
 		Length -= sizeof( m_DemoHeader );
 
 		if ( strcmp ( m_DemoHeader.demofilestamp, DEMO_HEADER_ID ) )
@@ -165,7 +168,7 @@ bool CDemoFile::Open( const char *name )
 		}
 
 		m_fileBuffer.resize( Length );
-		fread( &m_fileBuffer[ 0 ], 1, Length, fp );
+		read_size = fread( &m_fileBuffer[ 0 ], 1, Length, fp );
 
 		fclose( fp );
 		fp = NULL;

--- a/src/demofilebitbuf.cpp
+++ b/src/demofilebitbuf.cpp
@@ -688,6 +688,7 @@ void CBitRead::ReadBitAngles( QAngle& fa )
 float CBitRead::ReadBitFloat( void )
 {
 	uint32 nvalue = ReadUBitLong( 32 );
-	return *( ( float * ) &nvalue );
+	float fvalue = (float) nvalue;
+	return fvalue;
 }
 

--- a/src/demofiledump.cpp
+++ b/src/demofiledump.cpp
@@ -592,7 +592,6 @@ void ParseStringTableUpdate( CBitRead &buf, int entries, int nMaxEntries, int us
 	};
 
 	int lastEntry = -1;
-	int lastDictionaryIndex = -1;
 
 	// perform integer log2() to set nEntryBits
 	int nTemp = nMaxEntries;
@@ -660,7 +659,7 @@ void ParseStringTableUpdate( CBitRead &buf, int entries, int nMaxEntries, int us
 		unsigned char tempbuf[ MAX_USERDATA_SIZE ];
 		memset( tempbuf, 0, sizeof( tempbuf ) );
 		const void *pUserData = NULL;
-		int nBytes = 0;
+		unsigned int nBytes = 0;
 
 		if ( buf.ReadOneBit() )
 		{
@@ -725,7 +724,7 @@ void ParseStringTableUpdate( CBitRead &buf, int entries, int nMaxEntries, int us
 		{
 			if ( g_bDumpStringTables )
 			{
-				printf( " %d, %s, %d, %s \n", entryIndex, pEntry, nBytes, pUserData );
+				printf( " %d, %s, %d, %p \n", entryIndex, pEntry, nBytes, pUserData );
 			}
 		}
 
@@ -832,7 +831,7 @@ CSVCMsg_SendTable *GetTableByClassID( uint32 nClassID )
 {
 	for ( uint32 i = 0; i < s_ServerClasses.size(); i++ )
 	{
-		if ( s_ServerClasses[ i ].nClassID == nClassID )
+		if ( static_cast< uint32 >(s_ServerClasses[ i ].nClassID) == nClassID )
 		{
 			return &(s_DataTables[ s_ServerClasses[i].nDataTable ]);
 		}
@@ -1000,7 +999,7 @@ void FlattenDataTable( int nServerClass )
 			{
 				const CSVCMsg_SendTable::sendprop_t *prop = flattenedProps[currentProp].m_prop;
 
-				if (prop->priority() == priority || (priority == 64 && (SPROP_CHANGES_OFTEN & prop->flags()))) 
+				if (static_cast< uint32 >(prop->priority()) == priority || (priority == 64 && (SPROP_CHANGES_OFTEN & prop->flags()))) 
 				{
 					if ( start != currentProp )
 					{
@@ -1161,6 +1160,9 @@ void PrintNetMessage< CSVCMsg_PacketEntities, svc_PacketEntities >( CDemoFileDum
 		int nHeaderBase = -1;
 		int nNewEntity = -1;
 		int UpdateFlags = 0;
+
+		(void) nBaseline;
+		(void) bUpdateBaselines;
 
 		UpdateType updateType = PreserveEnt;
 
@@ -1451,7 +1453,8 @@ bool ParseDataTable( CBitRead &buf )
 	while ( 1 )
 	{
 		int type = buf.ReadVarInt32();
-		
+		(void)type;
+
 		void *pBuffer = NULL;
 		int size = 0;
 		if ( !ReadFromBuffer( buf, &pBuffer, size ) )

--- a/src/demofilepropdecode.h
+++ b/src/demofilepropdecode.h
@@ -121,6 +121,8 @@ struct Prop_t
 					printf( "%" PRIu64 "\n", m_value.m_int64 );
 				}
 				break;
+			case DPT_NUMSendPropTypes:
+				break;
 		}
 
 		if ( m_nNumElements > 1 )

